app/src/lumen/java/com/cliqz/browser/subscription/PurchaseFragment.kt

```diff
@@ -0,0 +1,110 @@
+import kotlinx.android.synthetic.lumen.fragment_purchase.*
+import javax.inject.Inject
+
+private val TAG = PurchaseFragment::class.java.simpleName
+
+class PurchaseFragment : DialogFragment(), OnBuyClickListener {
+
+    private lateinit var mAdapter: ProductListAdapter
+
+    @set:Inject
+    lateinit var premiumSubscriptionManager: PremiumSubscriptionManager
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(DialogFragment.STYLE_NORMAL, R.style.Theme_Purchase_Dialog)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? =
+            inflater.inflate(R.layout.fragment_purchase, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initRecyclerView()
+        setLoading(true)
+        getProductDetails()
+
+        BrowserApp.getActivityComponent(activity as MainActivity)?.inject(this)
+    }
+
+    private fun initRecyclerView() {
+        mAdapter = ProductListAdapter(null, this)
+        product_list.adapter = mAdapter
+        product_list.layoutManager = LinearLayoutManager(context)
+    }
+
+    private fun getProductDetails() {
+        Purchases.sharedInstance.getEntitlements(object : ReceiveEntitlementsListener {
+            override fun onReceived(entitlementMap: Map<String, Entitlement>) {
+                val productList = ArrayList<ProductRowData>()
+                entitlementMap[SubscriptionConstants.Entitlements.PREMIUM_SALE]?.offerings?.forEach {
+                    (_, offering) -> offering.skuDetails?.apply {
+                        productList.add(ProductRowData(sku, title, price, description))
+                    }
+                }
+                if (productList.isEmpty()) {
+                    // TODO: Display error
+                } else {
+                    mAdapter.updateProductList(productList)
+                    setLoading(false)
+                }
+            }
+
+            override fun onError(error: PurchasesError) {
+                Log.w(TAG, error.message)
+                // TODO: Display error
+            }
+        })
+    }
+
+    fun setLoading(flag: Boolean) {
+        product_list.visibility = if (flag) View.GONE else View.VISIBLE
+        loading.visibility = if (flag) View.VISIBLE else View.GONE
+    }
+
+    override fun onBuyClicked(position: Int) {
+        mAdapter.getProduct(position)?.apply {
+            Purchases.sharedInstance.makePurchaseWith(activity as Activity, sku,
+                    BillingClient.SkuType.SUBS,
+                    { error, _ ->
+                        Log.e(TAG, error.underlyingErrorMessage)
+                    },
+                    { _, purchaseInfo ->
+                        if (purchaseInfo.allPurchasedSkus.any { it.contains("vpn") }) {
```

Comment:
match full names and use constants

Reply:
Contains made sense here.

These are the products I use for staging:

com.cliqz.android.lumen.staging.sale.basic_vpn
com.cliqz.android.lumen.staging.sale.vpn
com.cliqz.android.lumen.staging.sale.basic

Even these are not the finalized identifiers for the products.

Reply:
> Even these are not the finalized identifiers for the products.
Thats why we should match full strings. So we don't have to change the code if "contains" logic doesnt stand valid anymore